sympy/matrices/matrices.py

```diff
@@ -390,16 +390,36 @@ def det(self, method="bareiss", iszerofunc=None):
 
 
         Possible values for "method":
-          bareis
+          bareiss
           berkowitz
           lu
         """
 
         # sanitize `method`
-        method = method.lower()
+        if not method.islower():
+            method_warn = method
+            method = method.lower()
+            SymPyDeprecationWarning(
+                feature="det(method='{}')".format(method_warn),
+                useinstead="det(method='{}')".format(method),
+                deprecated_since_version="1.4",
+                issue=15256
```

Comment:
This should be the number of a deprecation removal issue (see https://github.com/sympy/sympy/wiki/Deprecating-policy).